posthog/api/llm_prompt.py

```diff
@@ -0,0 +1,135 @@
+import re
+
+import posthoganalytics
+from rest_framework import serializers, status, viewsets
+from rest_framework.decorators import action
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from posthog.api.forbid_destroy_model import ForbidDestroyModel
+from posthog.api.routing import TeamAndOrgViewSetMixin
+from posthog.api.shared import UserBasicSerializer
+from posthog.models.llm_prompt import LLMPrompt
+from posthog.rate_limit import BurstRateThrottle, SustainedRateThrottle
+
+
+class LLMPromptSerializer(serializers.ModelSerializer):
+    created_by = UserBasicSerializer(read_only=True)
+
+    class Meta:
+        model = LLMPrompt
+        fields = [
+            "id",
+            "name",
+            "prompt",
+            "version",
+            "created_by",
+            "created_at",
+            "updated_at",
+            "deleted",
+        ]
+        read_only_fields = [
+            "id",
+            "version",
+            "created_by",
+            "created_at",
+            "updated_at",
+        ]
+
+    def validate_name(self, value: str) -> str:
+        if not re.match(r"^[a-zA-Z0-9_-]+$", value):
+            raise serializers.ValidationError(
+                "Only letters, numbers, hyphens (-) and underscores (_) are allowed.",
+                code="invalid_name",
+            )
+
+        return value
+
+    def validate(self, data):
+        team = self.context["get_team"]()
+        name = data.get("name")
+        existing_prompt = self.instance
+
+        # On CREATE: check if name already exists
+        if self.context["request"].method == "POST":
+            if LLMPrompt.objects.filter(name=name, team=team, deleted=False).exists():
+                raise serializers.ValidationError({"name": "A prompt with this name already exists."}, code="unique")
+
+        # On UPDATE: check if name changed OR if restoring a deleted prompt
+        if existing_prompt:
```

Comment:
I think the existing pattern, rather than checking on the request method, is usally something like:
```suggestion
        # On CREATE: check if name already exists
        if self.instance is None:
            if LLMPrompt.objects.filter(name=name, team=team, deleted=False).exists():
                raise serializers.ValidationError({"name": "A prompt with this name already exists."}, code="unique")

        # On UPDATE: check if name changed OR if restoring a deleted prompt
        else:
```